libs/langchain/langchain/chains/base.py

```diff
@@ -227,7 +227,9 @@ async def _acall(
             A dict of named outputs. Should contain all outputs specified in
                 `Chain.output_keys`.
         """
-        raise NotImplementedError("Async call not supported for this chain type.")
+        return await asyncio.get_running_loop().run_in_executor(
+            None, self._call, inputs, run_manager
```

Comment:
Are we able to remove the conditional from ainvoke so it's always relying on acall?

```
    async def ainvoke(
        self,
        input: Dict[str, Any],
        config: Optional[RunnableConfig] = None,
        **kwargs: Any,
    ) -> Dict[str, Any]:
        if type(self)._acall == Chain._acall:
            # If the chain does not implement async, fall back to default implementation
            return await asyncio.get_running_loop().run_in_executor(
                None, partial(self.invoke, input, config, **kwargs)
            )

```

Same question probably applies to a number of the other method implementations

Reply:
Yes would be good to do for all the classes that now have default async implementation